packages/snaps-controllers/src/snaps/location/npm.ts

```diff
@@ -186,7 +186,8 @@ export class NpmLocation implements SnapLocation {
         getNodeStream(tarballResponse),
         // The "gz" in "tgz" stands for "gzip". The tarball needs to be decompressed
         // before we can actually grab any files from it.
-        createGunzipStream(),
+        // To prevent recursion-based zip bombs, we set a maximum recursion depth of 3.
+        createGunzipStream(3),
```

Comment:
I don't know any non-malicious reason for having nested gzips in npm packages.
```suggestion
        createGunzipStream(1),
```

Reply:
Good point, `3` is the gunzip default, but I have verified that `1` isn't problematic.